core/controllers/suggestion.py

```diff
@@ -481,6 +481,12 @@ class SuggestionListHandler(base.BaseHandler):
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
 
+    URL_PATH_ARGS_SCHEMAS = {}
+
+    HANDLER_ARGS_SCHEMAS = {
```

Comment:
@Nik-09 @sahiljoster32  do you have guidance on how this should be handled? It seems to me that the handler references self.request.GET but with nondeterministic keys.

I think we might need to enumerate the keys in feconf.ALLOWED_SUGGESTION_QUERY_FIELDS here and let them have optional types but wanted to double-check.

Reply:
Hmm, this is a bit different case -- let me explain:

If you see [the failing tests](https://github.com/oppia/oppia/actions/runs/3149166949/jobs/5120486106#step:8:3170), then you can find that every test related to this handler is throwing the same error ( found extra args ) and this is because this handler can accept an arbitrary number of URL arguments that's why we are using `self.request.GET` in line 499 not `self.request.get` ( this is an exceptional case because I found `self.request.GET` in only this place in the codebase ). Now, for the validation part, we are already doing validation in line 502. but for schema validation, I really doubt we have a way to validate arbitrary number of URL args @Nik-09 can you please confirm?

Reply:
I dig into this and I think this is a bit different case, In my opinion, validation should be done in function body rather than by schema validation. @seanlip PTAL! ( Cc @Nik-09 )

Reply:
But I don't think this is a case where there's an arbitrary number of URL args. It should be possible to (a) look at the backend, (b) look at the frontend code that calls this handler, (c) enumerate all possible args that are seen in (a) and (b) (this should be not a very large list and there's no point passing stuff to the backend that's not used anyway) and overall just make this look like the other handlers.

@sahiljoster32 do you see anything that contradicts this?

Reply:
Sorry for the delay, yes, I agree with @seanlip, I think the solution mentioned above will work.

If in any case, you find the difficulty feel free to reach out to me or @sahiljoster32 we will help to find a solution for this.

Thank you.

Reply:
@Nik-09 It looks like it's just failing one test now: when there is an invalid query field. How would I be able to account for this in the schema?

Reply:
Yes, that error msg was expected.

[This check](https://github.com/oppia/oppia/blob/dfd054f830a75474aae9863e1f56aa43a9367cb5/core/controllers/suggestion.py#L556) is no more needed because the schema validation will now take care of invalid queries.

Reply:
@Nik-09 I took out the check, but now it looks like its failing an additional test. I read the logs but I can't really figure out what is messing up. could you help me out with this?